src/supervisor.tex

```diff
@@ -1092,6 +1092,14 @@ \subsection{Supervisor Memory-Management Fence Instruction}
       rs2}, except for entries containing global mappings.
 \end{itemize}
 
+If the value held in {\em rs1} is not a valid virtual address, then the
+SFENCE.VMA instruction has no effect.  No exception is raised in this case.
+
```

Comment:
The following non-normative comment seems inconsistent with the preceding normative statement.  The preceding requires that an SFENCE.VMA with an invalid VA has no effect (as in must have no effect).   Whereas the following comment suggests that such an SFENCE.VMA may in fact have an effect.  So which is the intention?  

The former seemingly would be preferable so that such an SFENCE.VMA could reliably be turned into an SFENCE.VMA that only has ordering effects.  But this assumes that some address (e.g. 0x7FFF_FFFF_FFFF_FFFF) will be invalid across all implementations, for now and forever.  Which seems pretty brittle (starting with wondering how J-ext pointer masking and memory tagging will interact).

So it seems like the above normative statement should be softened to indicate that "no effect" is an allowed behavior (since requiring "no effect" has no real benefit).

Reply:
I don't see a conflict here.  The specified behavior of SFENCE.VMA with invalid address is that it has no effect.

It is also true that SFENCE.VMA with invalid address may over-invalidate.  That's also true of any SFENCE.VMA.  It's also true in general: the implementation can invalidate address-translation cache entries any time it wants to, even when no SFENCE.VMA is executed.  That's an implementation detail at that point.  We shouldn't try to architecturally specify all the possible forms of over-invalidation.